relay.go

```diff
@@ -933,6 +933,7 @@ func (rfs *relayFragmentSender) flushFragment(wf *writableFragment) error {
 	sent, failure := rfs.frameReceiver.Receive(wf.frame, requestFrame)
 	if !sent {
 		rfs.failRelayItemFunc(rfs.outboundRelayItems, rfs.origID, failure, errFrameNotSent)
+		rfs.framePool.Release(wf.frame)
```

Comment:
I looked at ho other frameReceiver.Receive are handled. 
My understanding is that this flushFragment maintains the invariant that wf.frame is always released. The release is either by Receive() above or here. Correct?